encoder.go

```diff
@@ -0,0 +1,152 @@
+package schema
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+type encoderFunc func(reflect.Value) string
+
+// Encoder encodes values from a struct into url.Values.
+type Encoder struct {
+	cache  *cache
+	regenc map[reflect.Type]encoderFunc
+}
+
+// NewEncoder returns a new Encoder with defaults.
+func NewEncoder() *Encoder {
+	return &Encoder{cache: newCache(), regenc: make(map[reflect.Type]encoderFunc)}
+}
+
+// Encode encodes a struct into map[string][]string.
+//
+// Intended for use with url.Values.
+func (e *Encoder) Encode(src interface{}, dst map[string][]string) error {
```

Comment:
Sorry to be nitpicking again, but since this is meant to be used with `url.Values`, I think it's better to state that explicitly in the type instead of of the comment. dst could just be of type `url.Values` and the documentation updated accordingly.

This also makes the API more similar to github.com/google/go-querystring, though I prefer that the value is passed in like this instead of allocated internally like that library does.